revpi_core.c

```diff
@@ -63,6 +63,11 @@ u8 revpi_core_find_gate(struct net_device *netdev, u16 module_type)
 		goto ret_undef;
 	}
 
+	if (piDev_g.machine_type != REVPI_CONNECT_SE){
```

Comment:
What it this supposed to do? This should at least check against CORE_SE shouldnt it? Also at the end of the function we return with REV_PI_DEV_UNDEF anyway in case of CORE_SE or CONNECT_SE...

Reply:
> What it this supposed to do? This should at least check against CORE_SE shouldnt it? 
yes, it should be piDev_g.machine_type == REVPI_CORE_SE
> Also at the end of the function we return with REV_PI_DEV_UNDEF anyway in case of CORE_SE or CONNECT_SE...
actually on SE devices, this function should never be called, but on Core or Connect, with if statements it causes loss of performance. IMHO, normally the assert should be used here, but for Linux kernel module, it might be BUG_ON(...) or anything else?  Or nothing needs to be checked at all in this function.

Reply:
shouldn't the check of device type ` if (piDev_g.machine_type == REVPI_CONNECT_SE) {...}` be replace with BUG_ON(...), or just be removed?

Reply:
As I wrote above we return an error anyway, so I would just remove it

Reply:
> > What it this supposed to do? This should at least check against CORE_SE shouldnt it?
> 
> yes, it should be piDev_g.machine_type == REVPI_CORE_SE
> 
> > Also at the end of the function we return with REV_PI_DEV_UNDEF anyway in case of CORE_SE or CONNECT_SE...
> 
> actually on SE devices, this function should never be called, but on Core or Connect, with if statements it causes loss of performance. 

We really dont care for performance at this code path

Reply:
I just remove the check of device type in revpi_core_find_gate